src/EditorFeatures/CSharpTest/CodeActions/InvertIf/InvertIfTests.cs

```diff
@@ -0,0 +1,715 @@
+        {
+            await TestFixOneAsync(
+@"string x; [||]if (x.Length > 0x0) { a(); } else { b(); } } } ",
+@"string x; if (x.Length == 0x0) { b(); } else { a(); } } } ");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero4()
+        {
+            await TestFixOneAsync(
+@"string x; [||]if (0 < x.Length) { a(); } else { b(); } } } ",
+@"string x; if (0 == x.Length) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero5()
+        {
+            await TestFixOneAsync(
+@"byte x = 1; [||]if (0 < x) { a(); } else { b(); } } } ",
+@"byte x = 1; if (0 == x) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero6()
+        {
+            await TestFixOneAsync(
+@"ushort x = 1; [||]if (0 < x) { a(); } else { b(); } } } ",
+@"ushort x = 1; if (0 == x) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero7()
+        {
+            await TestFixOneAsync(
+@"uint x = 1; [||]if (0 < x) { a(); } else { b(); } } } ",
+@"uint x = 1; if (0 == x) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero8()
+        {
+            await TestFixOneAsync(
+@"ulong x = 1; [||]if (0 < x) { a(); } else { b(); } } } ",
+@"ulong x = 1; if (0 == x) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero9()
+        {
+            await TestFixOneAsync(
+@"ulong x = 1; [||]if (0 == x) { a(); } else { b(); } } } ",
+@"ulong x = 1; if (0 < x) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero10()
+        {
+            await TestFixOneAsync(
+@"ulong x = 1; [||]if (x == 0) { a(); } else { b(); } } } ",
+@"ulong x = 1; if (x > 0) { b(); } else { a(); } } } ");
```

Comment:
i don't ike this.  why did we convert ```==``` to ```>```  instead of ```!=```?

Reply:
Since unsigned numbers can't be negative, it can only be `> 0`.  For signed numbers you would get `!= 0` as expected.

Reply:
I get that that translation is "correct".  I'm saying i strictly do not like it :).  If i use == on an unsigned value, i still want != when negated.  I'm *ok* with moving in the other direction.  i.e. if i have  ```if (x > 0)``` then i'm ok with you converting that to ```if (x == 0)```.  

Note: this is specifically about unsigned types.  I'm totally ok with special logic around Array/String.Length.